### DISCUSSION TITLE
How to Access and Modify wp-config.php via FTP on AWS, Backup WordPress Site, and Update Plugins

### DISCUSSION THREAD

**Discussion by adfs3njj** [community/community#127539] [Programming Help]
### Select Topic Area

Question

### Body

I need to:

- Access the wp-config.php file via FTP (SSH) using Filezilla, WinSCP, or Ubuntu Linux on a VM.
- Toggle the logic from define('DISALLOW_FILE_MODS', true); to define('DISALLOW_FILE_MODS', false);.
- Save the changes and reboot the AWS server.
- Backup the theme and all data using the WP Data Migration tool (which we are currently paying for).
- Update the WordPress core and all plugins.

**Background**

I work at a charity and handle UX and content support. Seven years ago, the developer who built our site restricted access, preventing us from editing the theme or updating plugins. Here are some key points for context:

- The developer deliberately locked down our privileges.
- Other developers have refused to help without proper access rights, suggesting a complete site rebuild instead.
- The original development team offered to find the AWS key for a significant fee.

We need to regain access to our site to perform updates and backups without resorting to a costly rebuild.

1. AWS EC2 Management Console:

- Set up inbound traffic for my IP, allowing all traffic and SSH in Security Groups.

2. Putty and PuttyGen:

- Followed steps from this YouTube tutorial on managing WordPress files on AWS with FileZilla FTP.https://www.youtube.com/watch?v=XoO0ca-UHUA

Created key pairs in AWS, converted .pem to .ppk file in PuttyGen, and configured settings in Putty.

Followed setup on Filezilla

Essentially I am not able to connect to the AWS instance.

![putty error 1](https://github.com/community/community/assets/171936037/390f10c8-7476-46c2-998d-d324ecdc971f)
![putty setup 1](https://github.com/community/community/assets/171936037/c197c95a-ad02-44f3-8a7d-8ec8a3f2699f)
![putty setup 1](https://github.com/community/community/assets/171936037/18c83852-3337-46a1-a5c1-d080172ddcc0)

**Comment by LiteBrite82** [comment]
Thanks for posting in the GitHub Community, @adfs3njj ! 

We’ve moved your post to our Programming Help 🧑‍💻 category, which is more appropriate for this type of discussion.

Please review our [guidelines](https://github.com/orgs/community/discussions/51384) about the Programming Help category for more information.

**Comment by Hadikamali** [comment]
Hi

It seems to me that there is an issue with public key authentication, which means that the key pair may not be set up correctly.

-You are using the correct key pair that matches the one associated with your AWS EC2 instance.
-Convert the .pem file to a .ppk file using PuTTYgen if you haven't done so already.

**Comment by kathlene-cory** [comment]
Hi @adfs3njj - did @Hadikamali's solution resolve this problem or do you still need assistance?
